File: frontend/src/api/settingService.js
import axiosClient from './axiosClient';

const settingService = {
  // Public
  getShippingSetting: async () => {
    const response = await axiosClient.get('/api/settings/shipping');
    return response.data;
  },

  getGeneralSetting: async () => {
    const response = await axiosClient.get('/api/settings/general');
    return response.data;
  },

  // Admin
  getAdminShippingSetting: async () => {
    const response = await axiosClient.get('/api/admin/settings/shipping');
    return response.data;
  },

  updateShippingSetting: async (data) => {
    const response = await axiosClient.put('/api/admin/settings/shipping', data);
    return response.data;
  },

  getAdminGeneralSetting: async () => {
    const response = await axiosClient.get('/api/admin/settings/general');
    return response.data;
  },

  updateGeneralSetting: async (data) => {
    const response = await axiosClient.put('/api/admin/settings/general', data);
    return response.data;
  },
};

export default settingService;

File: mobile/lib/services/setting_service.dart
import 'package:dio/dio.dart';
import '../core/constants/api_constants.dart';
import '../core/network/dio_client.dart';
import '../models/banner_model.dart';
import '../models/setting_model.dart';

class SettingService {
  final Dio _dio = DioClient().dio;

  /// Gets active banners for promotional slider
  Future<List<BannerModel>> getActiveBanners() async {
    final response = await _dio.get(ApiConstants.banners);
    final list = response.data as List<dynamic>? ?? [];
    return list.map((e) => BannerModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  /// Gets general shop branding and settings
  Future<ShopGeneralSettingModel> getGeneralSetting() async {
    try {
      final response = await _dio.get(ApiConstants.generalSettings);
      return ShopGeneralSettingModel.fromJson(response.data as Map<String, dynamic>);
    } catch (_) {
      return ShopGeneralSettingModel.defaultSetting();
    }
  }

  /// Gets shipping configuration (shipping fee & free ship threshold)
  Future<ShippingSettingModel> getShippingSetting() async {
    try {
      final response = await _dio.get(ApiConstants.shippingSettings);
      return ShippingSettingModel.fromJson(response.data as Map<String, dynamic>);
    } catch (_) {
      return ShippingSettingModel.defaultSetting();
    }
  }
}
